### PR TITLE
Use Java 11

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Use Java 11, not 8, to let us run this project with Scala Steward (it's [currently failing.](https://github.com/guardian/scala-steward-public-repos/actions/runs/5765604057/job/15631890372#step:5:3590))

## How to test

Deploy to CODE. All work as expected? Be sure to test both the management and checker services, perhaps by adding a new rule, publishing it, and testing it in Composer.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -8,7 +8,7 @@ ThisBuild / version := "1.0-SNAPSHOT"
 ThisBuild / scalacOptions := Seq(
   "-encoding",
   "UTF-8",
-  "-target:jvm-1.8",
+  "-target:jvm-11",
```
Comment (rtyley) [inline build.sbt:11]: Small nit, but `-target` got deprecated with https://github.com/scala/scala/pull/9982, now `-release` is preferred!

```suggestion
  "-release:11",
```

See https://github.com/guardian/frontend/commit/8405bfc2775469119e31f2b0264ab8ccbc9af0a5 for another example of this.

**Review by rtyley — APPROVED** [review]
I haven't run it, but looks good to me, thank you! 👍

File: build.sbt (hunk 1 above)
Comment (jonathonherbert) [reply build.sbt:11]: Thanks @rtyley, TIL!

Comment (rtyley) [reply build.sbt]: Ah, apologies, I see that with 090b9c899b367d08ce1b71a6ea0688a62c5fb578 you've addressed the fact that the `-release` flag wasn't added with https://github.com/scala/scala/pull/9982 until Scala v2.13.9, and this repo was on Scala 2.13.7! Sorry, didn't mean to delay you...

Comment (jonathonherbert) [reply build.sbt]: No worries – could have chosen to defer, but felt it was worth bumping anyhow!

**Comment by rtyley** [comment]
Thanks! I can confirm that the Public Repos Scala Steward job is [now passing](https://github.com/guardian/scala-steward-public-repos/actions/runs/5799037515#:~:text=typerighter) 🚀 

image

image

**Comment by jonathonherbert** [comment]
Thanks for following up, @rtyley 🙏

**Pull request merged**